components/script/fetch.rs

```diff
@@ -261,6 +261,7 @@ impl FetchResponseListener for FetchContext {
 
     fn process_response_chunk(&mut self, mut chunk: Vec<u8>) {
         self.body.append(&mut chunk);
+        self.response_object.root().stream_chunk(chunk.as_slice());
```

Comment:
It turns out that `self.body.append(&mut chunk);` empties the passed vector, which is why a bunch of tests that I would expect to pass are failing with these changes. If we reverse the order of these lines, then the `compileStreaming` tests from /wasm/webapi/contenttype.any.html (and likely other test files as well) should start passing.

Reply:
I have made the changes. Now the mimeType related tests are passing. However, some of the response status tests in ```/tests/wpt/metadata/wasm/webapi/status.any.js.ini``` are failing, which I believe should have passed. Is this expected?